hicommonwealth/commonwealth
# Fix walletconnect-Terra

## Link to Issue
Related: #6332 
Closes: #6408 

## Description of Changes
- Fixes WalletConnect for /terra
- Reduces terra bundle size by 20%, but couldn't get rid of terra.js - even though it's not referenced in code anymore, `@terra-money/wallet-provider` depends on it
- Migration adds a ChainNode RPC for Terra2 to match the cosmos_chain_id of "terra2." This allows us to use `cosmjs` to `getRecentBlock` instead of terra.js (which uses an LCD endpoint).

## "How We Fixed It"

Bundle before:
![master](https://github.com/hicommonwealth/commonwealth/assets/9438198/f34609e2-cd0d-41d3-81b5-b15e6e6ef222)

Bundle after:
![mybranch](https://github.com/hicommonwealth/commonwealth/assets/9438198/99650a1b-7c22-4942-96cd-6f75257095b4)


## Test Plan
- run migration
- CA (click around) tested on local and frick:
  - go to /terra and sign in with WalletConnect (and Station Mobile)
    - should work via desktop QR modal
    - should work via mobile browser 
  -  Sign in with regular desktop and Station flow - should still work

## Deployment Plan

1. 

## Other Considerations

- https://github.com/terra-money/wallet-provider does not list terra.js as a dependency, but our app wouldn't build without it.
